tests/functions.php

```diff
@@ -1,3 +1,168 @@
 <?php
+require_once'../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class Functions extends TestCase
+{
+    public function testSuccessValidateWeaponName()
+    {
+        $expected = true;
+        $input = ['name' => 'Firestorm', 'damage' => 10];
+        $case = validateWeaponName($input);
+        $this->assertEquals($expected,$case);
+    }
+
+    public function testMalformedValidateWeaponName()
+    {
+        $input = 'name';
+        $this->expectException(TypeError::class);
+        validateWeaponName($input);
+    }
+
+    public function testSuccessValidateWeaponData()
+    {
+        $expected = true;
+        $input = [[
+            'name' => 'Firestorm',
+            'damage_per_second' => '5',
+            'rate_of_fire' => '5',
+            'ammo_consumption' => '4',
+            'ammo_efficiency' =>  '2',
+            'default_weapon_number' => '3',
+            'image_url' => 'some kind of url'
+            ]];
+        $case = validateWeaponData($input);
+        $this->assertEquals($expected,$case);
+    }
+    public function testSuccess2ValidateWeaponData()
```

Comment:
This should probably be named testFailureValidateWeaponData(). Could add a second example/assertion using an input array with missing keys for extra rigour.

Reply:
see above

Reply:
I don't think this is a failure test. I'd consider any input array as a proper use of this function